File: examples/java/greetings_project/greeting_lib/tests/com/example/GreetingTest.java
package com.example;

import org.junit.Test;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import static com.google.common.truth.Truth.assertThat;

public class GreetingTest {
   @Test
    public void testGreetNoNames() {
        List<String> got = Greeting.getGreetings("greetings", Collections.emptyList());
       assertThat(got).isEmpty();
    }

    @Test
    public void testGreetOneName() {
        List<String> got = Greeting.getGreetings("greetings", Collections.singletonList("name"));
        assertThat(got).containsExactly("greetings name!");
    }

    @Test
    public void testGreetMultipleNames() {
        List<String> got = Greeting.getGreetings("greetings", Arrays.asList("name1", "name2"));
        assertThat(got).containsExactly("greetings name1!", "greetings name2!");
    }
}

File: examples/kotlin/simple_project/test/lib/GreetingTest.kt
package test.lib

import com.google.common.truth.Truth.assertThat
import main.lib.Greeting
import org.junit.Test

internal class GreetingTest {

  @Test
  fun testGetNoGreetings() {
    val actual = Greeting.getGreetings(arrayOf())
    assertThat(actual).isEmpty()
  }

  @Test
  fun testGetSingleGreeting() {
    val actual = Greeting.getGreetings(arrayOf("Usersss"))
    assertThat(actual).asList().containsExactly("Hello Usersss!")
  }

  @Test
  fun testGetMultipleGreetings() {
    val actual = Greeting.getGreetings(arrayOf("User 1", "User 2", "User 3"))
    val expected = arrayOf("Hello User 1!", "Hello User 2!", "Hello User 3!")
    assertThat(actual).asList().containsExactlyElementsIn(expected)
  }
}
